docs/src/main/asciidoc/metrics-guide.adoc

```diff
@@ -0,0 +1,180 @@
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+@Path("/")
+public class PrimeNumberChecker {
+
+    private long highestPrimeNumberSoFar = 3;
+
+    @GET
+    @Path("/{number}")
+    @Produces("text/plain")
+    @Counted(name = "performedChecks", monotonic = true, description = "How many primality checks have been performed.")
+    @Timed(name = "checksTimer", description = "A measure how long it takes to perform the primality test.", unit = MetricUnits.MILLISECONDS)
+    public String checkIfPrime(@PathParam("number") long number) {
+        if (number < 1) {
+            return "Only natural numbers can be prime numbers.";
+        }
+        if (number == 1) {
+            return "1 is not prime.";
+        }
+        if (number % 2 == 0) {
+            return number + " is not prime, it is divisible by 2.";
+        }
+        for (int i = 3; i < Math.floor(Math.sqrt(number)) + 1; i = i + 2) {
+            if (number % i == 0) {
+                return number + " is not prime, is divisible by " + i + ".";
+            }
+        }
+        if (number > highestPrimeNumberSoFar) {
+            highestPrimeNumberSoFar = number;
+        }
+        return number + " is prime.";
+    }
+
+    @Gauge(name = "highestPrimeNumberSoFar", unit = MetricUnits.NONE, description = "Highest prime number so far.")
+    public Long highestPrimeNumberSoFar() {
+        return highestPrimeNumberSoFar;
+    }
+
+}
+----
+
+== Running and using the application
+
+To run the microservice:
+
+* To run in dev mode, use `./mvnw clean compile quarkus:dev`
+
+=== Generate some values for the metrics
+First, ask the endpoint whether some numbers are prime numbers.
+
+ curl localhost:8080/350
+
+The application will respond that 350 is not a prime number because it can be divided by 2.
+
+Now for some large prime number so that the test takes a bit more time:
+
+ curl localhost:8080/629521085409773
+
+The application will respond that 629521085409773 is a prime number.
+If you want, try some more calls with numbers of your choice.
+
+=== Review the generated metrics
+To view the metrics, execute `curl -H"Accept: application/json" localhost:8080/metrics/application`
+You will receive a response such as:
+
+```
+{
+  "org.acme.metrics.PrimeNumberChecker.checksTimer" : {
+    "p50": 217.231273,
+    "p75": 217.231273,
+    "p95": 217.231273,
+    "p98": 217.231273,
+    "p99": 217.231273,
+    "p999": 217.231273,
+    "min": 0.58961,
+    "mean": 112.15909190834819,
+    "max": 217.231273,
+    "stddev": 108.2721053982776,
+    "count": 2,
+    "meanRate": 0.04943519091742238,
+    "oneMinRate": 0.2232140583080189,
+    "fiveMinRate": 0.3559527083952095,
+    "fifteenMinRate": 0.38474303050928976
+  },
+  "org.acme.metrics.PrimeNumberChecker.performedChecks" : 2,
+  "org.acme.metrics.PrimeNumberChecker.highestPrimeNumberSoFar" : 629521085409773
+}
+```
+
+Description what each metric means:
```

Comment:
`These metrics are worth an explanation:` would be better I think

Reply:
All of them are explained there, not just some. In some cases, multiple values are just squashed into one line to make it shorter.

Reply:
Yeah, agreed. I'm just reformulating what you're saying. You can use `Those` or `All those` if you prefer :).

Reply:
Or `Let's explain what each metric means:`